xds/src/test/java/io/grpc/xds/internal/certprovider/CommonCertProviderTestUtils.java

```diff
@@ -52,4 +56,25 @@ private static String getResourceContents(String resourceName) throws IOExceptio
     }
     return text;
   }
+
+  /** Allow tests to register a provider using test clock.
+  */
+  public static void register(final FakeClock fakeClock) {
+    FileWatcherCertificateProviderProvider tmp = new FileWatcherCertificateProviderProvider(
+        FileWatcherCertificateProvider.Factory.getInstance(),
+        new ScheduledExecutorServiceFactory() {
+
+            @Override
+            ScheduledExecutorService create() {
+              return fakeClock.getScheduledExecutorService();
+            }
+          },
+          TimeProvider.SYSTEM_TIME_PROVIDER);
+    CertificateProviderRegistry.getInstance().register(tmp);
+  }
+
+  public static void register0() {
```

Comment:
What's the purpose of `register0()`. Instead of doing this can you have `register()` return `tmp` (the registered object) and define an `unregister` that takes that value to unregister from the registry? You can then use register/unregister in each test

Reply:
The `FileWatcherCertificateProviderProvider`  has a static name. So calling `register()` with a fake clock overrides the default [FileWatcherCertificateProviderProvider](https://github.com/grpc/grpc-java/blob/bfcba82dd5c142ec0078c514c3ec28c583fb41d7/xds/src/main/java/io/grpc/xds/internal/certprovider/CertificateProviderRegistry.java#L42), and calling `deregister` would make the registry empty, that is why we restore it to default with `register0`, and other test cases can go through.
Use a setup() :`register` and teardown(): deregister pattern does not really fit here I believe.

Reply:
Yes, good point.